seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/snowflake/SnowflakeTypeMapper.java

```diff
@@ -0,0 +1,138 @@
+    private static final String SNOWFLAKE_DOUBLE_PRECISION = "DOUBLE PRECISION";
+    private static final String SNOWFLAKE_REAL = "REAL";
+
+    private static final String SNOWFLAKE_VARCHAR = "VARCHAR";
+    private static final String SNOWFLAKE_CHAR = "CHAR";
+    private static final String SNOWFLAKE_CHARACTER = "CHARACTER";
+    private static final String SNOWFLAKE_STRING = "STRING";
+    private static final String SNOWFLAKE_TEXT = "TEXT";
+    private static final String SNOWFLAKE_BINARY = "BINARY";
+    private static final String SNOWFLAKE_VARBINARY = "VARBINARY";
+
+    private static final String SNOWFLAKE_BOOLEAN = "BOOLEAN";
+
+    private static final String SNOWFLAKE_DATE = "DATE";
+    private static final String SNOWFLAKE_DATE_TIME = "DATE_TIME";
+    private static final String SNOWFLAKE_TIME = "TIME";
+    private static final String SNOWFLAKE_TIMESTAMP = "TIMESTAMP";
+    private static final String SNOWFLAKE_TIMESTAMP_LTZ = "TIMESTAMPLTZ";
+    private static final String SNOWFLAKE_TIMESTAMP_NTZ = "TIMESTAMPNTZ";
+    private static final String SNOWFLAKE_TIMESTAMP_TZ = "TIMESTAMPTZ";
+
+    private static final String SNOWFLAKE_GEOGRAPHY = "GEOGRAPHY";
+    private static final String SNOWFLAKE_GEOMETRY = "GEOMETRY";
+
+    private static final String SNOWFLAKE_VARIANT = "VARIANT";
+    private static final String SNOWFLAKE_OBJECT = "OBJECT";
+
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex)
+            throws SQLException {
+        String snowflakeType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        int scale = metadata.getScale(colIndex);
+        switch (snowflakeType) {
+            case SNOWFLAKE_SMALLINT:
+            case SNOWFLAKE_TINYINT:
+            case SNOWFLAKE_BYTEINT:
+                return BasicType.SHORT_TYPE;
+            case SNOWFLAKE_INTEGER:
+            case SNOWFLAKE_INT:
+                return BasicType.INT_TYPE;
+            case SNOWFLAKE_BIGINT:
+                return BasicType.LONG_TYPE;
+            case SNOWFLAKE_DECIMAL:
+            case SNOWFLAKE_NUMERIC:
+            case SNOWFLAKE_NUMBER:
+                return new DecimalType(precision, scale);
+            case SNOWFLAKE_REAL:
+            case SNOWFLAKE_FLOAT4:
+                return BasicType.FLOAT_TYPE;
+            case SNOWFLAKE_DOUBLE:
+            case SNOWFLAKE_DOUBLE_PRECISION:
+            case SNOWFLAKE_FLOAT8:
+            case SNOWFLAKE_FLOAT:
+                return BasicType.DOUBLE_TYPE;
+            case SNOWFLAKE_BOOLEAN:
+                return BasicType.BOOLEAN_TYPE;
+            case SNOWFLAKE_CHAR:
+            case SNOWFLAKE_CHARACTER:
+            case SNOWFLAKE_VARCHAR:
+            case SNOWFLAKE_STRING:
+            case SNOWFLAKE_TEXT:
+            case SNOWFLAKE_VARIANT:
+            case SNOWFLAKE_OBJECT:
+            case SNOWFLAKE_GEOMETRY:
```

Comment:
In Snowflake, to convert `SNOWFLAKE_GEOMETRY` and `SNOWFLAKE_GEOGRAPHY` into EWKT format strings, you need to set session-level parameters (GEOGRAPHY_OUTPUT_FORMAT='EWKT', GEOMETRY_OUTPUT_FORMAT='EWKT').

Reference: https://docs.snowflake.com/en/sql-reference/data-types-geospatial

Reply:
It should either be added to the URL or execute the SQL `GEOGRAPHY_OUTPUT_FORMAT='EWKT' `directly. I suggest trying to add it to the URL first to see if it takes effect.

Reply:
`?GEOGRAPHY_OUTPUT_FORMAT=EWKT` this does work in url.
<img width="405" alt="Screen Shot 2023-05-12 at 11 43 22 PM" src="https://github.com/apache/incubator-seatunnel/assets/20422210/62c09f9b-3096-42cf-95f3-c06ac9fc43d1">
<img width="282" alt="Screen Shot 2023-05-12 at 11 43 33 PM" src="https://github.com/apache/incubator-seatunnel/assets/20422210/b1e2250e-310b-4213-9248-2321c836aafc">

Reply:
I've updated in type mapper: https://github.com/apache/incubator-seatunnel/blob/98f4ecf7b7219eca141ae111a89e9e1db9804e13/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/snowflake/SnowflakeTypeMapper.java#L117,
let me know if this makes sense to you

Reply:
This makes a lot of sense, let's add it to the code and documentation

Reply:
thx, updated